### ISSUE TITLE
API for wait(...)

### ISSUE THREAD

**Issue by Trial-In-Error** [testing-library/pptr-testing-library#12]
How exactly does wait() work in pttr-testing-library? The examples in the readme appear to be un-usable. For instance, when I attempt to use syntax like that given in the readme:

```
await wait(() => getByText('Loading...'))
```

I receive a rejection with: `containerHandle.executionContext is not a function`.

Ok, so maybe I need to provide a 1st argument context to getByText:
```
const $document = await getDocument(page)
await wait(() => getByText($document, 'Loading...'))
```

That rejects with: `TypeError: Cannot read property 'defaultView of null'`.

Ok... Well, getByText returns a promise, right? Seems like I should pend on the returned promise (why should it behave differently in the browser than in node?):
```
const $document = await getDocument(page)
await wait(async () => await getByText($document, 'Loading...'))
```

That also rejects with: `TypeError: Cannot read property 'defaultView of null'`.

Hm... Well, the context inside the `wait` is likely the browser, so maybe it can't find my $document variable from the outer scope?:
```
await wait(async () => {
    const $document = await getDocument(page)
    await getByText($document, 'Loading...')
})
```

That also rejects with: `TypeError: Cannot read property 'defaultView of null'`.

Ok, well, what can I learn about the context inside the wait?
```
await wait(() => {
    console.log(getDocument)
    console.log(getByText)
}
```

This logs into the terminal (not the browser) and has access to variables in the enclosing scope. So my theory about the execution context being the browser is wrong. Why isn't it able to find the defaultView property? What is the correct syntax for `wait`?

**Comment by patrickhulce** [comment]
Thanks for filing @Trial-In-Error! What version of puppeteer and `pptr-testing-library` are you using? I haven't updated in a while and puppeteer internals may have changed on me :)

**Comment by Trial-In-Error** [comment]
I'm currently on pptr-testing-library@0.3.2, puppeteer@1.16.0

**Comment by patrickhulce** [comment]
Hm standard tests are passing on that version though `/extends` is indeed broken (see #13).

I'll have to take a deeper look tomorrow.

**Comment by patrickhulce** [comment]
Thanks for reporting this was indeed a bug based on interaction between our lagging wait-for-expect and puppeteer versions. Added tests for multiple versions to catch these in the future and published the fix in v0.4.0
